src/Castle.Facilities.AspNetCore/Contributors/CrossWiringComponentModelContributor.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright 2004-2018 Castle Project - http://www.castleproject.org/
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Castle.Facilities.AspNetCore.Contributors
+{
+	using System;
+
+	using Castle.Core;
+	using Castle.MicroKernel;
+	using Castle.MicroKernel.ModelBuilder;
+	using Castle.Facilities.AspNetCore.Activators;
+
+	using Microsoft.Extensions.DependencyInjection;
+
+	public class CrossWiringComponentModelContributor : IContributeComponentModelConstruction
+	{
+		private readonly IServiceCollection services;
+
+		public CrossWiringComponentModelContributor(IServiceCollection services)
+		{
+			this.services = services;
+		}
+
+		public void ProcessModel(IKernel kernel, ComponentModel model)
+		{
+			if (model.Configuration.Attributes.Get("aspnet-core-cross-wired") == "yes")
```

Comment:
I know some of the facilities don't but it would be good to use a constant for `aspnet-core-cross-wired`, consider even making a [public readonly field like the typed factory](https://github.com/castleproject/Windsor/blob/bbc1183a8806f4cfdf07dc462e1895313fb0684a/src/Castle.Windsor/Facilities/TypedFactory/TypedFactoryFacility.cs#L39-L48).

Reply:
Hint accepted. I promised a first class API(excluding typed factories and parent child containers). LOL.

Reply:
( Now I know enough to do this really easy :) )

Reply:
There is no need for an `IFacility` implementing class, just find a sensible place for the constant, probably `WindsorRegistrationExtensions`.